include/unifex/sync_wait.hpp

```diff
@@ -125,27 +105,18 @@ using receiver_t = typename _receiver<T>::type;
 
 template <typename Result, typename Sender>
 std::optional<Result> _impl(Sender&& sender) {
-  auto blockingResult = blocking(sender);
-  const bool completesSynchronously =
-      blockingResult == blocking_kind::always ||
-      blockingResult == blocking_kind::always_inline;
-
   using promise_t = _sync_wait::promise<Result>;
   promise_t promise;
-  if (!completesSynchronously) {
-    promise.doneEvent_.emplace();
-  }
+  manual_event_loop ctx;
```

Comment:
Yes, that occurred to me.

Reply:
I've made a note on the P1898R1 PR to constrain this in the text.

As more than Receivers can be scheduler_providers maybe we have to define it in terms of the lifetime of the provider, rather than the set_ calls though.

Reply:
Yes, I think defining the validity of the scheduler in terms of being limited to the lifetime of the provider seems like it might generalise to other use-cases.